web/modules/custom_weather_module/src/Plugin/Block/CustomWeatherBlock.php

```diff
@@ -37,30 +38,40 @@ public function build() {
       '#theme' => 'weather_theme',
       '#weather_header' => 'Weather now in ',
       '#user_location' => $this->userLocation(),
-      '#conditions' => $this->userWeather(),
-      '#icon' => 'https://cdn.weatherapi.com/weather/64x64/day/113.png',
+      '#conditions' => $this->userWeather()['conditions'],
+      '#icon' => $this->userWeather()['icon'],
```

Comment:
What happens if `$this->userWeather()` method fails?
It is much better practice to check the result - to avoid the "Notice undefined offset" messages and/or Null pointer exception.

Do not forget that your method always should return the value to easy handling the result of work. 
Hint: you can return NULL from the method and handle it with the '??' (Null Coalescing Operator)